### DISCUSSION TITLE
How to reuse AsyncClient fixture

### DISCUSSION THREAD

**Discussion by michaelxswe** [fastapi/fastapi#10833] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from contextlib import asynccontextmanager
from fastapi import FastAPI
import asyncio
from httpx import AsyncClient
import pytest

@asynccontextmanager
async def lifespan(app: FastAPI):
    print("starting up")
    yield
    print("shutting down")

app = FastAPI(lifespan=lifespan)

@app.get("/hello")
async def hello():
    await asyncio.sleep(2)
    return {"hello": "world"}

@pytest.fixture(scope="session")
async def client():
    async with lifespan(app) as test_app:
        async with AsyncClient(app=test_app, base_url="http://localhost") as client:
            yield client

@pytest.mark.asyncio
async def test_hello(client):
    response = await client.get("/hello")
    assert response.json() == {"hello": "world"}
```

### Description

I'm trying to reuse AsyncClient,  and it seems that the fixture is not properly injected into the test_hello function.

Running pytest will result in the following error message:

```
client = <async_generator object client at 0x00000175B3EA1990>
    @pytest.mark.asyncio
    async def test_hello(client):
>       response = await client.get("/hello")
E       AttributeError: 'async_generator' object has no attribute 'get'
```

### Operating System

Windows

### FastAPI Version

0.104.1

### Pydantic Version

2.5.1

### Python Version

3.11.1

**Comment by Ventura94** [comment, marked as answer]
@michaelxswe 
The fixed code is the following, look at the comments:
````python
import asyncio
from contextlib import asynccontextmanager

import pytest
from fastapi import FastAPI
from httpx import AsyncClient


@asynccontextmanager
async def lifespan(app: FastAPI):
    print("starting up")
    yield
    print("shutting down")


app = FastAPI(lifespan=lifespan)


@app.get("/hello")
async def hello():
    await asyncio.sleep(2)
    return {"hello": "world"}


# conftest.py
@pytest.fixture(scope="session") # I add the following fixture to configure asyncio in pytest
def anyio_backend():
    return "asyncio"


@pytest.fixture(scope="session")
async def client():
    async with lifespan(app):  # lifespan does not return the asgi app
        async with AsyncClient(app=app, base_url="http://localhost") as client:
            yield client


# test_hello.py
@pytest.mark.anyio  # change @pytest.mark.asyncio by @pytest.mark.anyio
async def test_hello(client: AsyncClient):
    response = await client.get("/hello")
    assert response.json() == {"hello": "world"}

````
I hope this code solves your problem.
